### PR TITLE
Move MarketingEmailClient and ProfileClient to user.js

### PR DESCRIPTION
**Pull request by kritisingh1**
As per the demands of the issue #4983 , I have moved the functions MarketingEmailClient and ProfileClient from app-start.js to user.js. @vladikoff please review my PR #5473  and suggest required changes. Thanks a lot.

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 40aa23b9376f0fa0141aa1fb72934bb7d0fd0e41
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 4d6cce1504f5c0d37e613ad318ac8e97065079cd
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 555b33696b0607f73f26d267b2d389c921068bc2
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: f7d3b65cc5e8ab2ec98a444787ab99cd0c1bf5eb
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: a50f72c35c3dd940397a773da66258766110595e
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

File: app/scripts/lib/app-start.js [hunk 1]
```diff
@@ -33,14 +33,14 @@ define(function (require, exports, module) {
   const HeightObserver = require('lib/height-observer');
   const IframeChannel = require('lib/channels/iframe');
   const InterTabChannel = require('lib/channels/inter-tab');
-  const MarketingEmailClient = require('lib/marketing-email-client');
+  /*const MarketingEmailClient = require('lib/marketing-email-client');*/
```
Comment (vladikoff) [inline app/scripts/lib/app-start.js]: All the commented out code needs to be deleted 👍

**Comment by vladikoff** [comment]
@kritisingh1 This is a good start!

File: app/scripts/models/user.js [hunk 2]
```diff
@@ -81,6 +83,30 @@ define(function (require, exports, module) {
       return this._storage.get('accounts') || {};
     },
 
+    initializeDep() {
+      return p()
+
```
Comment (vladikoff) [inline app/scripts/models/user.js]: remove this new line

File: app/scripts/lib/app-start.js (hunk 1 above)
Comment (kritisingh1) [reply app/scripts/lib/app-start.js]: Sure @vladikoff

File: app/scripts/models/user.js [hunk 3]
```diff
@@ -81,6 +83,30 @@ define(function (require, exports, module) {
       return this._storage.get('accounts') || {};
     },
 
+    initializeDep() {
```
Comment (vladikoff) [inline app/scripts/models/user.js]: @kritisingh1 where is this new function called ?

File: app/scripts/models/user.js [hunk 4]
```diff
@@ -23,6 +23,8 @@ define(function (require, exports, module) {
   const SearchParamMixin = require('models/mixins/search-param');
   const Storage = require('lib/storage');
   const vat = require('lib/vat');
+  const MarketingEmailClient = require('lib/marketing-email-client');
+  const ProfileClient = require('lib/profile-client');
 
   var User = Backbone.Model.extend({
     initialize (options = {}) {
```
Comment (vladikoff) [inline app/scripts/models/user.js]: `marketingEmailClient, profileClient` are currently passed in as `options`, but you just moved those two things into this file. Now in this file those two things need to be initialized. So you need to modify, 

```
this._marketingEmailClient = options.marketingEmailClient;
``` 

and 

```
this._profileClient = options.profileClient;
``` 

below

File: app/scripts/models/user.js (hunk 3 above)
Comment (kritisingh1) [reply app/scripts/models/user.js]: I think I need to make another function which calls initializeDep() in user.js itself. I am so sorry I didn't notice this.

Comment (vladikoff) [reply app/scripts/models/user.js]: Actually you don't need this extra function

File: app/scripts/models/user.js (hunk 4 above)
Comment (vladikoff) [reply app/scripts/models/user.js]: Instead take the code that you added below and do something like:

```
      this._profileClient = new ProfileClient({
        profileUrl: this._config.profileUrl
      });
```

and a similar thing for `marketingEmailClient`.

One thing though, you don't have access to `this._config` on this file. so you need to pass it in as an option to user.js

File: app/scripts/models/user.js (hunk 3 above)
Comment (vladikoff) [reply app/scripts/models/user.js]: See https://github.com/mozilla/fxa-content-server/pull/5473/files#r138177383

File: app/scripts/models/user.js (hunk 4 above)
Comment (kritisingh1) [reply app/scripts/models/user.js]: So, after initialising it that way, I need to remove those two functions , namely ````initializeProfileClient```` and ````initializeMarketingClient```` , right?

Comment (vladikoff) [reply app/scripts/models/user.js]: @kritisingh1 Correct!

Comment (vladikoff) [reply app/scripts/models/user.js]: @kritisingh1 You can do most of it inside the `initialize` function (of user.js)

**Comment by vladikoff** [comment]
@kritisingh1 Feel free to reopen a new PR with the fixes listed above! 👍

**Pull request closed without merging**